runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -753,9 +764,9 @@ private int decodeInitialType(
         if (packet != null)
         {
             final int length = packet.remainingLength();
-            final MqttPacketType packetType = MqttPacketType.valueOf(packet.typeAndFlags() >> 4);
+            final MqttPacketType packetType = MqttPacketType.valueOf((packet.typeAndFlags() >> 4) * 2 + server.typeOffset);
 
-            if (packetType != MqttPacketType.CONNECT)
+            if (packetType != MqttPacketType.CONNECTV5 && packetType != MqttPacketType.CONNECTV4)
```

Comment:
This `decodeInitialType` can become the generic `decodeConnect` to determine the version, then reject or set decoder to `decodePacketTypeV4` or `decodePacketTypeV5`.